### DISCUSSION TITLE
wsl vscode esp-idf flashing to usb device

### DISCUSSION THREAD

**Discussion by skealy** [dorssel/usbipd-win#393] [Q&A]
I have successfully tested building the Hello-world example on a WSL Ubuntu-20.04 linux in Visual Studio Code.
I have attached the usb using usbipd wsl attach --busid=1-8. (1-8 is the busid obtained from the usbipd wsl list command).
In a ESP-IDF terminal I can build "idf.py build" the project successfully. I can also flash the esp32 successfully using the "idf.py -p /dev/ttyUSB0 flash" command in the ESP-IDF terminal.
If I use the Command Palette "Build your project" option the project builds successfully. If I use the Command 
Palette "Flash (UART) your project" option it fails with Permission Error(13, 'Access is denied', None, 5).
I attach a file containing the Doctor output and the terminal output which includes the error.
[USBerror.txt](https://github.com/dorssel/usbipd-win/files/8771231/USBerror.txt)
Thanks in advance to anybody with helpful suggestions!

**Comment by dorssel** [comment]
> serial.serialutil.SerialException: could not open port '/dev/ttyUSB0': PermissionError(13, 'Access is denied.', None, 5)

This usually means that you do not have the correct `udev` rule for the device and/or that the `udev` service was not started. See also: https://github.com/dorssel/usbipd-win/wiki/WSL-support#udev.
